src/main/java/com/valstro/plugin/validator/FieldValidator.java

```diff
@@ -53,31 +59,52 @@ public void validate(Class<?> clazz, Class<?> codeGenClazz) throws MojoFailureEx
         }
     }
 
+    private void checkEnumValues(final Class<?> clazzType) throws MojoFailureException {
+        var codeGen = this.holder.getCodeGenClassFromName(clazzType.getSimpleName());
+        var enumValuesClass = clazzType.getEnumConstants();
+
+        Set<String> enumValues = Arrays.stream(codeGen.getEnumConstants())
```

Comment:
**note**: I know the existence of EnumSet, however for some reason does not like, this is the reason why I moved to this solution